sdk/core/System.ClientModel/src/Options/ClientLoggingOptions.cs

```diff
@@ -0,0 +1,277 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Exposes client options for logging within a <see cref="ClientPipeline"/>.
+/// </summary>
+public class ClientLoggingOptions
+{
+    private bool _frozen;
+
+    private bool? _enableLogging;
+    private bool? _enableMessageLogging;
+    private bool? _enableMessageContentLogging;
+    private int? _messageContentSizeLimit;
+    private ILoggerFactory? _loggerFactory;
+    private PipelineMessageSanitizer? _sanitizer;
+    private bool? _allowedHeaderNamesHasChanged;
+    private bool? _allowedQueryParametersHasChanged;
+    private IList<string>? _allowedHeaderNames;
+    private IList<string>? _allowedQueryParameters;
+
+    internal const bool DefaultEnableLogging = true;
+    internal const bool DefaultEnableMessageContentLogging = false;
+
+    internal const double RequestTooLongSeconds = 3.0; // sec
+    internal const int DefaultMessageContentSizeLimit = 4 * 1024;
+    internal static string[] DefaultAllowedHeaderNames { get; } = new[] {
+            "traceparent",
+            "Accept",
+            "Cache-Control",
+            "Connection",
+            "Content-Length",
+            "Content-Type",
+            "Date",
+            "ETag",
+            "Expires",
+            "If-Match",
+            "If-Modified-Since",
+            "If-None-Match",
+            "If-Unmodified-Since",
+            "Last-Modified",
+            "Pragma",
+            "Request-Id",
+            "Retry-After",
+            "Server",
+            "Transfer-Encoding",
+            "User-Agent",
+            "WWW-Authenticate" };
+
+    internal static string[] DefaultAllowedQueryParameters { get; } = new[] { "api-version" };
+
+    internal static PipelineMessageSanitizer DefaultSanitizer { get; } = new(DefaultAllowedHeaderNames, DefaultAllowedQueryParameters);
```

Comment:
Debated on if this should live here or in the PipelineMessageSanitizer class, chose here since it is dependent on the defaults in the options